### PR TITLE
DPC-4258 pass if add tests

### PR DESCRIPTION
**Pull request by jdettmannnava**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4258

## 🛠 Changes

ci-workflow updated to find tests added to project
- Paths in .resultset.json edited to point to absolute paths in docker container that runs sonarqube
- Unit tests run for api
- Jacoco results from build-api used for api verification

## ℹ️ Context

The original implementation only involved making sure the workflow failed if code was outed without tests. This update makes sure the workflow passes if new code is tested.

## 🧪 Validation

Changes tested in https://github.com/CMSgov/dpc-app/pull/2276, which has the same workflow file. It also has new code and tests in dpc-api, dpc-portal, dpc-web, and dpc-admin.

### CODE REVIEW

**Comment by chris-ronning-ny** [comment]
When you merge main in, make sure to revert the changes I made [here](https://github.com/CMSgov/dpc-app/pull/2267/files#diff-e42b2b9794d0333e41c8e218ece5897ea7395746b5e07d08fb85b53e36d15043R138) in order to unblock my other PR.

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -44,14 +64,14 @@ jobs:
       - name: "DPC Web Build"
         run: |
           make ci-web-portal
-      - name: "Move the test results" # won't upload hidden files
+      - name: "Reformat test results" # Sonarqube will run in a docker container and wants the paths to be from /github/workspace
         run: |
-          sudo mv ./dpc-web/coverage/.resultset.json ./dpc-web/coverage/resultset.json
+          sudo jq '.RSpec.coverage |= with_entries(if .key | contains("dpc-web") then .key |= sub("/dpc-web"; "/github/workspace/dpc-web") else . end)' dpc-web/coverage/.resultset.json > web-resultset.json
```
Comment (MEspositoE14s) [inline .github/workflows/ci-workflow.yml:69]: I didn't realize `jq` could do more than pretty print json!

**Review by MEspositoE14s — APPROVED** [review]
LGTM 👍

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (jdettmannnava) [reply .github/workflows/ci-workflow.yml:69]: This is what the Jenkins step did...

**Pull request merged**